src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -86,6 +89,33 @@ namespace OpenMS
     std::vector<double> normalized_library_intensity;
     getNormalized_library_intensities_(transitions, normalized_library_intensity);
 
+    // If add_up_spectra_ is -1, we automatically compute the amount of spectra to add by setting add_up_spectra_ to the ceiling value of N percent of the retention time peak width
+    if (add_up_spectra_ == -1)
+    {
+        double leftWidth = imrmfeature->getMetaValue("leftWidth");
+        double rightWidth = imrmfeature->getMetaValue("rightWidth");
+        double peakWidth = rightWidth - leftWidth;
+        add_up_spectra_ = std::ceil(peakWidth * use_percent_peak_width_);
+        
+        // Ensure add_up_spectra_ is odd
+        if (static_cast<int>(add_up_spectra_) % 2 == 0) {
+            add_up_spectra_ -= 1;
+        }
+        
+        // Ensure add_up_spectra_ is at least 1 if it was set to <= 0 based on the peak width
```

Comment:
I think this is dangerous. Changing the value that represents something stored in the internal param object can easily lead to unexpected and hard to debug bugs. Imagine (this arificial example) that e.g., another thread would update a totally unrelated parameter of the Param object. This would call updateParameters of that class (which might set add_up_spectra_ to 0 again).
It would be really hard to find bugs related to add_up_spectra_ suddenly being 0.
Also, I think values other than -1 don't have a semantic meaning here. So the correct way would be to fail for all non-semantically valid values.

Reply:
so it can never be -1? after this line?

Reply:
This is a good point. I think your comment below for setting a local variable would be better than modifying the param object. 

The actual value for add_up_spectra needs to be a positive value, otherwise `fetchSpectrumSwath` will fail. I use -1 to identify whether `add_up_spectra` should be computed dynamically, but maybe that would cause some downstream errors?

Reply:
Should I add another arg to to turn on the auto `add_up_spectra`? Or change `add_up_spectra` to be a string option?

Reply:
I would probably not change it into a string.
e.g. one could define:
```
enum class SpectrumMergeMethodType { Fixed, Dynamic };
and change the initialize function to take both the SpectrumMergeMethodType and a double merge_window_size
```
and then query this information in the calculateDIAScores method.
Depending on the SpectrumMergeMethodType this will fill the local variable e.g. n_merge_spectra with a constant value or dynamic by peak width.

Reply:
> I would probably not change it into a string. e.g. one could define:
> 
> ```
> enum class SpectrumMergeMethodType { Fixed, Dynamic };
> and change the initialize function to take both the SpectrumMergeMethodType and a double merge_window_size
> ```
> 
> and then query this information in the calculateDIAScores method. Depending on the SpectrumMergeMethodType this will fill the local variable e.g. n_merge_spectra with a constant value or dynamic by peak width.

Okay, I have implemented something along these lines now.